metaserver/src/main/scala/scala/meta/languageserver/search/SymbolIndex.scala

```diff
@@ -111,6 +112,21 @@ class SymbolIndex(
     }
   }
 
+  /** Returns the definition position of the symbol at the given position */
+  def references(
+      path: AbsolutePath,
+      line: Int,
+      column: Int
+  ): ReferencesResult = {
+    val locations = for {
+      symbol <- findSymbol(path, line, column).toSeq
+      (uri, ranges) <- symbol.references
+      range <- ranges.ranges
+    } yield i.Position(uri, Some(range)).toLocation
```

Comment:
@olafurpg could you take a look at this piece of code? I'm probably doing it wrong, but this way I think we won't get references from the index because it stores only global symbols, i.e. not their usage which is what we want for references. 

Currently I get _some_ references only for fields/parameters. `User` for example doesn't give any results.

Reply:
Could it be that you're missing fallback symbols with `alternatives(symbol)`? `example.User#` is the type symbol and `example.User.` is the companion object term symbol. They refer to different things, but if the term companion is synthetic then you need to fallback to it's type symbol.

We probably need to refine the `alternatives` method since definition and references have different rules for how alternatives are computed, but for now I guess it's fine to use the same implementation.

The index currently stores all definitions and references to all symbols. Soon we'll probably stop storing definitions for global symbols and references to local symbols since those can easily be computed on the fly from a `Document`.